include/mbedtls/x509_csr.h

```diff
@@ -213,6 +222,21 @@ void mbedtls_x509write_csr_set_md_alg( mbedtls_x509write_csr *ctx, mbedtls_md_ty
  *                  set using this function.
  */
 int mbedtls_x509write_csr_set_key_usage( mbedtls_x509write_csr *ctx, unsigned char key_usage );
+#undef MBEDTLS_DEPRECATED
+#endif /* !MBEDTLS_DEPRECATED_REMOVED */
+
+/**
+ * \brief           Set the Key Usage Extension flags
+ *                  (e.g. MBEDTLS_X509_KU_DIGITAL_SIGNATURE |
+ *                  MBEDTLS_X509_KU_KEY_CERT_SIGN).
+ *
+ * \param ctx       CSR context to use.
+ * \param key_usage Key usage flags to set.
+ *
+ * \return          0 if successful, or MBEDTLS_ERR_X509_ALLOC_FAILED
```

Comment:
If you don't mind, we could improve the Doxygen style along the way here (as it has been done in many other modules some time ago), always using full sentences and splitting the documentation for multiple return values into separate `\return` sections.

Reply:
Also, I think we should fall back to the generic `A negative error code on failure` statement about the failing case - at least we can return `MBEDTLS_ERR_X509_FEATURE_UNAVAILABLE` now.

Reply:
Sure, will do that. I essentially just copy-pasted the documentation for mbedtls_x509write_csr_set_key_usage() and updated the relevant parts.